frontend/apps/web/components/jobs/NosqlTable/NosqlTable.tsx

```diff
@@ -230,9 +262,23 @@ function AddNewRecord(props: AddNewRecordProps): ReactElement {
     context: {
       accountId: account?.id,
       isUserJavascriptCodeValid: validateUserJsCodeAsync,
+      data: data,
```

Comment:
If we could not pass the data directly into the context as that is a pretty large object.
Let's keep the context simply by instead passing in a function that can be called by the context to check.

If you see my comments below about writing a single `isduplicate` function, that may be able to be either used here 1:1 or augments to handle both with shared logic.